package.json

```
@@ -77,7 +78,7 @@
     "node-dir": "0.1.17",
     "prettier": "2.6.2",
     "react": "17.0.2",
-    "react-docgen": "^5.0.0-beta.1",
+    "react-docgen": "rnmapbox/react-docgen#rnmapbox",
```

```suggestion
    "react-docgen": "6.0.0-alpha.3",
```

I think it now contains the fixes we need See [592](https://github.com/reactjs/react-docgen/pull/592)

---

So if I integrate this, imported types (e.g. `AtmosphereLayerStyleProps` in `Atmosphere.tsx`) still don't expand in the documentation, and methods on the ref don't appear in `component.methods` anymore. Any idea how you want to handle this?

---

I just realized this is still open: https://github.com/reactjs/react-docgen/pull/591. Would you want to rebase and we point to your fork? Or I would be happy to set up a patch.

Would really like to just get this merged 😭 🤣

---

I've updated the PR and also pushed the merged version to: https://github.com/rnmapbox/react-docgen/tree/basic-hook-support. But last time I was checking it did not generate methods, because our structure is different to what's supported, but worth a try.